### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by alexandresablayrolles**
Removing drop_last as it is incompatible with a sampler + updating DDPOptimizer for batchsplitter.

### CODE REVIEW

**Review by ffuuugor — APPROVED** [review]

File: opacus/optimizers/ddpoptimizer.py [hunk 1]
```diff
@@ -47,7 +46,8 @@ def reduce_gradients(self):
                 p.grad /= self.world_size
 
     def step(self, closure: Optional[Callable[[], float]] = None) -> Optional[float]:
```
Comment (romovpa) [inline opacus/optimizers/ddpoptimizer.py:48]: Are we sure the output should be `float` and not `Tensor`? 

In this [example](https://pytorch.org/docs/stable/optim.html):
```python
for input, target in dataset:
    def closure():
        optimizer.zero_grad()
        output = model(input)
        loss = loss_fn(output, target)
        loss.backward()
        return loss
    optimizer.step(closure)
```

Here `loss` is obviously a `Tensor`, we can access grad and call `backward()`.

Comment (alexandresablayrolles) [reply opacus/optimizers/ddpoptimizer.py]: Fixed, thanks!
